File: app/src/main/java/com/tifone/tnews/base/IModel.java
package com.tifone.tnews.base;

import com.tifone.tnews.bean.home.HomeTestBean;

import java.util.List;

/**
 * Created by tongkao.chen on 2018/5/3.
 */

public interface IModel {
     interface OnDataLoadedCallback<T> {
          void onDataLoadedComplete(T data);
          void onDataLoadedFail(String reason);
     }
     void loadData(OnDataLoadedCallback<HomeTestBean> callback);
}

File: app/src/main/kotlin/com.tifone.tnews/base/IModel.kt
package com.tifone.tnews.base

import com.tifone.tnews.bean.home.HomeTestBean

interface IModel {
    interface OnDataLoadedCallback<T> {
        fun onDataLoadedComplete(data: T)
        fun onDataLoadedFail(reason: String)
    }

    fun loadData(callback: OnDataLoadedCallback<HomeTestBean>)
}